### ISSUE TITLE
Caches cleanup logic uses buggy version of commons-io

### ISSUE THREAD

**Issue by ALikhachev** [gradle/gradle#28868]
### Current Behavior

During clean up of the caches Gradle breaks permissions of the symlinked files, if it removes a directory that contains symlinks

### Expected Behavior

Gradle does not modify anything it should not :sligh

### Context (optional)

We detected it by using an artifact transformation that [creates symlinks](https://github.com/JetBrains/kotlin/blob/4722294bd5234b1af49909fc2a0b8b32cff03f6f/repo/gradle-build-conventions/buildsrc-compat/src/main/kotlin/test-symlink-transformation.gradle.kts) to dependencies to test executions against classpath containing some special characters, like whitespaces.

### Steps to Reproduce

Basically, the problem is reproducible quite easy:
1. Declare a dependency to `commons-io:commons-io:2.11.0`
2. Create a file, e.g. `touch test.txt`
3. Ensure it has permissions bitmask 644 (rw-r--r--)
4. Create a directory with a symlink to the file (e.g. `mkdir sym && ln -s ../test.txt sym/test.txt`)
5. Use commons-io to delete the directory (`FileUtils.deleteQuietly(File("sym"))` (that's what Gradle does)
6. The symlinked file now has permissions bitmask 444 (r--r--r--)

This is a bug of commons-io 2.11.0 https://issues.apache.org/jira/browse/IO-751 that was fixed in 2.12.0

### Gradle version

8.7

**Comment by asodja** [comment]
This should already be fixed with https://github.com/gradle/gradle/pull/28474, where we updated version to 2.15.1.

If there is still an issue let us know by adding a comment on this issue or open a new one.
